### ISSUE TITLE
Convert - .bim to .pbit

### ISSUE THREAD

**Issue by jasonhendrix** [pbi-tools/pbi-tools#150]
Is it possible to use the convert command (or a series of other commands) to convert a .bim file (or a Tabular Editor folder) to a .pbit or .pbix file?

If so, can you please give an example of what that command line would look like?

**Comment by 7samat** [comment]
As far as I am aware, it is not fully available to be scripted out-of-the-box; It might be so if you prepare a template with 3 extra items.

First step is to convert the bim to pbi-tools format of a folder tree:
`pbi-tools convert ".\Model.bim" ".\Dataset\Model" -overwrite`

If you are updating an existing report this way, you can skip to last step now. If you want to create a report-dataset file from scratch, there are a few things to add.

Add a pbix.proj.json:
`pbi-tools init Deployments .\Dataset`

Then you need to add 3 more items externally:

- ReportSettings.json
- ReportMetadata.json
- Version.txt

You can obtain them by saving any placeholder pbix file and running the standard pbi-tools extract.

Finally, you combine that into a PBI template file:
pbi-tools compile ".\Dataset" -format pbit -outPath "Dataset.pbit" -overwrite

**Comment by 7samat** [comment]
That's great, thanks for sharing. So far I only used a bim for quick edits in Tabular Editor, comimg back to pbi-tools folder structure after I'm done.  Specifically, 1 case was to enclose it in a ps script to temporarily convert, run a TE script and then recompile. I will need to test this tip and simplify my approach ;)

**Comment by jasonhendrix** [comment]
Thanks for your help!  I was able to compile to .PBIT format.

When I open the PBIT in Power BI Desktop though, I get this error:
Can't visit DataSource

Clicking view details on the error shows this detail:
PowerBINonFatalError:
{"AppName":"PBIDesktop","AppVersion":"2.105.1143.0","ModuleName":"Microsoft.PowerBI.Modeler.dll","Component":"Microsoft.PowerBI.Modeler.DefaultObjectTypeVisitor`1","Error":"System.NotImplementedException","MethodDef":"VisitDefault","ErrorOffset":"35"}

I also upgraded Desktop to the latest release from yesterday and tried again with the same result:
PowerBINonFatalError:
{"AppName":"PBIDesktop","AppVersion":"2.106.883.0","ModuleName":"Microsoft.PowerBI.Modeler.dll","Component":"Microsoft.PowerBI.Modeler.DefaultObjectTypeVisitor`1","Error":"System.NotImplementedException","MethodDef":"VisitDefault","ErrorOffset":"35"}

When I open the PBIT in Tabular Editor everything about Data Sources looks reasonable there.

**Comment by jasonhendrix** [comment]
Thanks for your help!

I was able to work through that issue and several other similar issues by removing the shared data sources, updating the PQ statements to not reference the shared data sources, and removing other incompatible items (role members, incremental refresh policies, etc.).

I'm having two other issues:
1. The tables all appear to be set to "DirectQuery" mode by default instead of "Import" even though everything was "Import" mode in the original .bim file.  Is there a way to avoid that?
2. In an attempt to find a scriptable way to fix the above issue, I tried to open the pbit file in Tabular Editor 2.16.7 but I get the error "This file is not a valid PBIX / PBIT file."
